test/object-store/sync/flx_migration.cpp

```diff
@@ -441,8 +441,12 @@ TEST_CASE("Test client migration and rollback with recovery", "[sync][flx][flx m
     //  Migrate back to FLX - and keep the realm session open
     trigger_server_migration(session.app_session(), MigrateToFLX, logger_ptr);
 
-    REQUIRE(!wait_for_upload(*outer_realm));
-    REQUIRE(!wait_for_download(*outer_realm));
+    // wait for the subscription store to initialize after downloading
```

Comment:
Instead of polling, one other way to fix this (I think) is to commit a change and keep the completion callbacks.